nikola/plugins/command/auto/__init__.py

```diff
@@ -0,0 +1,302 @@
+        reload_notifier = pyinotify.ThreadedNotifier(reload_wm, self.do_refresh)
+        reload_notifier.start()
+        reload_wm.add_watch(out_folder, MASK, rec=True)  # Watch output folders
+
+        # Start watchers that trigger rebuilds
+        rebuild_wm = pyinotify.WatchManager()
+        rebuild_notifier = pyinotify.ThreadedNotifier(rebuild_wm, self.do_rebuild)
+        rebuild_notifier.start()
+        for p in watched:
+            if os.path.exists(p):
+                rebuild_wm.add_watch(p, MASK, rec=True)  # Watch input folders
+
+        parent = self
+
+        class Mixed(WebSocketWSGIApplication):
+            """A class that supports WS and HTTP protocols in the same port."""
+            def __call__(self, environ, start_response):
+                if environ.get('HTTP_UPGRADE') is None:
+                    return parent.serve_static(environ, start_response)
+                return super(Mixed, self).__call__(environ, start_response)
+
+        ws = make_server(
+            host, port, server_class=WSGIServer,
+            handler_class=WebSocketWSGIRequestHandler,
+            app=Mixed(handler_cls=LRSocket)
+        )
+        ws.initialize_websockets_manager()
+        self.logger.info("Serving on port {0}...".format(port))
+
+        # Yes, this is racy
+        if browser:
+            webbrowser.open('http://{0}:{1}'.format(host, port))
+
+        try:
+            ws.serve_forever()
+        except KeyboardInterrupt:
+            ws.server_close()
+
+    def do_rebuild(self, event):
+        p = subprocess.Popen(self.command_line, shell=True, stderr=subprocess.PIPE)
+        if p.wait() != 0:
+            error = p.stderr.read()
+            self.logger.error(error)
+            error_signal.send(error=error)
+
+    def do_refresh(self, event):
+        self.logger.info('REFRESHING: ', event.pathname)
+        p = os.path.relpath(event.pathname, os.path.abspath(self.site.config['OUTPUT_FOLDER']))
+        refresh_signal.send(path=p)
+
+    def serve_static(self, environ, start_response):
+        """Trivial static file server."""
+        uri = wsgiref.util.request_uri(environ)
+        p_uri = urlparse(uri)
+        f_path = os.path.join(self.site.config['OUTPUT_FOLDER'], *p_uri.path.split('/'))
+        mimetype = mimetypes.guess_type(uri)[0] or b'text/html'
+
+        if os.path.isdir(f_path):
+            f_path = os.path.join(f_path, self.site.config['INDEX_FILE'])
+
+        if os.path.isfile(f_path):
+            with open(f_path) as fd:
+                start_response(b'200 OK', [(b'Content-type', mimetype)])
+                return self.inject_js(mimetype, fd.read())
+        elif p_uri.path == '/livereload.js':
+            with open(LRJS_PATH) as fd:
+                start_response(b'200 OK', [(b'Content-type', mimetype)])
+                return self.inject_js(mimetype, fd.read())
+        start_response(b'404 ERR', [])
+        return ['404 {0}'.format(uri)]
```

Comment:
Why not inject live reload here also? User could be fiddling with their slugs or other path manipulation.